core/storage/couchStorage.go

```diff
@@ -0,0 +1,1801 @@
+	removedDestinationPolicyServices := []common.ServiceID{}
+	return nil, removedDestinationPolicyServices, nil
+}
+
+// UpdateRemovedDestinationPolicyServices update the removedDestinationPolicyServices, only for ESS
+func (store *CouchStorage) UpdateRemovedDestinationPolicyServices(orgID string, objectType string, objectID string, destinationPolicyServices []common.ServiceID) common.SyncServiceError {
+	return nil
+}
+
+// UpdateNotificationRecord updates/adds a notification record to the object
+func (store *CouchStorage) UpdateNotificationRecord(notification common.Notification) common.SyncServiceError {
+	id := getNotificationCollectionID(&notification)
+	if notification.ResendTime == 0 {
+		resendTime := time.Now().Unix() + int64(common.Configuration.ResendInterval*6)
+		notification.ResendTime = resendTime
+	}
+
+	existingObject := &couchNotificationObject{}
+	if err := store.getOne(id, existingObject); err != nil {
+		if err != notFound {
+			return &Error{fmt.Sprintf("Failed to update notification record. Error: %s.", err)}
+		}
+		existingObject = nil
+	}
+
+	newObject := couchNotificationObject{ID: id, Notification: notification}
+	if existingObject != nil {
+		newObject.Rev = existingObject.Rev
+	}
+	err := store.upsertObject(id, newObject)
+	if err != nil {
+		return &Error{fmt.Sprintf("Failed to update notification record. Error: %s.", err)}
+	}
+	return nil
+}
+
+// UpdateNotificationResendTime sets the resend time of the notification to common.Configuration.ResendInterval*6
+func (store *CouchStorage) UpdateNotificationResendTime(notification common.Notification) common.SyncServiceError {
+	id := getNotificationCollectionID(&notification)
+	resendTime := time.Now().Unix() + int64(common.Configuration.ResendInterval*6)
+
+	result := &couchNotificationObject{}
+	if err := store.getOne(id, result); err != nil {
+		return &Error{fmt.Sprintf("Failed to update notification resend time. Error: %s.", err)}
+	}
+
+	result.Notification.ResendTime = resendTime
+	if err := store.upsertObject(id, result); err != nil {
+		return &Error{fmt.Sprintf("Failed to update notification resend time. Error: %s.", err)}
+	}
+	return nil
+}
+
+// RetrieveNotificationRecord retrieves notification
+func (store *CouchStorage) RetrieveNotificationRecord(orgID string, objectType string, objectID string, destType string,
+	destID string) (*common.Notification, common.SyncServiceError) {
+	id := createNotificationCollectionID(orgID, objectType, objectID, destType, destID)
+	result := couchNotificationObject{}
+	if err := store.getOne(id, &result); err != nil {
+		if err == notFound {
+			return nil, nil
+		}
+		return nil, &Error{fmt.Sprintf("Failed to fetch the notification. Error: %s.", err)}
+	}
+	return &result.Notification, nil
+}
+
+// DeleteNotificationRecords deletes notification records to an object
+//very different
```

Comment:
Is this comment necessary? I think you should either remove this line, or add more information to explain the difference(preferred)
(I assume you mean the implementation of this function has big difference from mongoStorage.go)

Reply:
Oops, sorry about this. I added this while implementing it for my own reference because the DeleteAll function is quite different. But this comment is not needed here, I will remove it